command_line/find_bad_pixels.py

```diff
@@ -133,29 +94,84 @@ def run(args):
         else:
             total += peak_pixels.as_1d().as_int()
 
-    p.finished("Finished finding hot pixels on %d images" % len(images))
+    return total
+
+
+def run(args):
+    usage = "dev.dials.find_bad_pixels [options] data_master.h5"
+
+    parser = OptionParser(
+        usage=usage,
+        phil=phil_scope,
+        read_experiments=True,
+        read_experiments_from_images=True,
+        epilog=help_message,
+    )
+
+    params, options = parser.parse_args(show_diff_phil=True)
+
+    experiments = flatten_experiments(params.input.experiments)
+    if len(experiments) != 1:
+        parser.print_help()
+        sys.exit("Please pass an experiment list\n")
+        return
+
+    imagesets = experiments.imagesets()
+
+    if len(imagesets) != 1:
+        sys.exit("Please pass an experiment list that contains one imageset")
+
+    imageset = imagesets[0]
+    detectors = imageset.get_detector()
+    assert len(detectors) == 1
+    detector = detectors[0]
+    trusted = detector.get_trusted_range()
+
+    first, last = imageset.get_scan().get_image_range()
+    images = range(first, last + 1)
+
+    if params.images is None and params.image_range is not None:
+        start, end = params.image_range
+        params.images = list(range(start, end + 1))
+
+    if params.images:
+        if min(params.images) < first or max(params.images) > last:
+            sys.exit("Image outside of scan range")
+        images = params.images
+
+    # work around issues with HDF5 and multiprocessing
+    imageset.reader().nullify_format_instance()
+
+    n = int(math.ceil(len(images) / params.nproc))
+    chunks = [images[i : i + n] for i in range(0, len(images), n)]
+    assert len(images) == sum([len(chunk) for chunk in chunks])
```

Comment:
```suggestion
    assert len(images) == sum(len(chunk) for chunk in chunks)
```
avoid explicit list construction